sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/Count.java

```diff
@@ -106,30 +118,70 @@ public void processElement(ProcessContext c) {
   /**
    * A {@link CombineFn} that counts elements.
    */
-  private static class CountFn<T> extends CombineFn<T, Long, Long> {
+  private static class CountFn<T> extends CombineFn<T, long[], Long> {
+    // Note that the long[] accumulator always has size 1, used as
+    // a box for a mutable long.
 
     @Override
-    public Long createAccumulator() {
-      return 0L;
+    public long[] createAccumulator() {
+      return new long[] {0};
     }
 
     @Override
-    public Long addInput(Long accumulator, T input) {
-      return accumulator + 1;
+    public long[] addInput(long[] accumulator, T input) {
+      accumulator[0] += 1;
+      return accumulator;
     }
 
     @Override
-    public Long mergeAccumulators(Iterable<Long> accumulators) {
+    public long[] mergeAccumulators(Iterable<long[]> accumulators) {
       long result = 0L;
```

Comment:
Consider using the following pattern (similar as BinaryCombineFn):
      Iterator<long[]> iter = accumulators.iterator();
      if (!iter.hasNext()) {
        return createAccumulator();
      }
      long[] running = iter.next();
      while (iter.hasNext()) {
        running[0] += iter.next()[0];
      }
      return running;